### ISSUE TITLE
Concatenating two empty Spectra fails

### ISSUE THREAD

**Issue by meowcat** [rformassspectrometry/Spectra#200]
```
library(Spectra)
spd <- DataFrame(msLevel = c(1L, 2L), rtime = c(1.1, 1.2))
spd$mz <- list(c(100, 103.2, 104.3, 106.5), c(45.6, 120.4, 190.2))
spd$intensity <- list(c(200, 400, 34.2, 17), c(12.3, 15.2, 6.8))
data <- Spectra(spd)
data_empty <- data[c()]
data_merged_1 <- c(data, data_empty) # works
data_merged_2 <- c(data_empty, data) # works
data_merged_3 <- c(data_empty, data_empty) # fails:
# Error in objects[[1]] : subscript out of bounds
```

**Comment by meowcat** [comment]
```
- Session info ----------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 4.0.4 (2021-02-15)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       Europe/Berlin               
 date     2021-04-19                  

- Packages --------------------------------------------------------------------------------------------------------------------------------------
 package      * version date       lib source                                           
 assertthat     0.2.1   2019-03-21 [2] CRAN (R 4.0.3)                                   
 backports      1.2.1   2020-12-09 [2] CRAN (R 4.0.3)                                   
 BiocGenerics * 0.36.1  2021-04-16 [1] Bioconductor                                     
 BiocParallel * 1.24.1  2020-11-06 [2] Bioconductor                                     
 broom          0.7.4   2021-01-29 [2] CRAN (R 4.0.3)                                   
 cachem         1.0.4   2021-02-13 [2] CRAN (R 4.0.4)                                   
 callr          3.5.1   2020-10-13 [2] CRAN (R 4.0.3)                                   
 cellranger     1.1.0   2016-07-27 [2] CRAN (R 4.0.3)                                   
 cli            2.3.0   2021-01-31 [2] CRAN (R 4.0.3)                                   
 clue           0.3-59  2021-04-16 [1] CRAN (R 4.0.4)                                   
 cluster        2.1.0   2019-06-19 [3] CRAN (R 4.0.4)                                   
 colorspace     2.0-0   2020-11-11 [2] CRAN (R 4.0.3)                                   
 crayon         1.4.1   2021-02-08 [2] CRAN (R 4.0.3)                                   
 DBI            1.1.1   2021-01-15 [2] CRAN (R 4.0.3)                                   
 dbplyr         2.1.0   2021-02-03 [2] CRAN (R 4.0.3)                                   
 desc           1.2.0   2018-05-01 [2] CRAN (R 4.0.3)                                   
 devtools     * 2.3.2   2020-09-18 [2] CRAN (R 4.0.3)                                   
 dplyr        * 1.0.4   2021-02-02 [2] CRAN (R 4.0.3)                                   
 ellipsis       0.3.1   2020-05-15 [2] CRAN (R 4.0.3)                                   
 fastmap        1.1.0   2021-01-25 [2] CRAN (R 4.0.3)                                   
 forcats      * 0.5.1   2021-01-27 [2] CRAN (R 4.0.3)                                   
 fs             1.5.0   2020-07-31 [2] CRAN (R 4.0.3)                                   
 generics       0.1.0   2020-10-31 [2] CRAN (R 4.0.3)                                   
 ggplot2      * 3.3.3   2020-12-30 [2] CRAN (R 4.0.3)                                   
 glue           1.4.2   2020-08-27 [2] CRAN (R 4.0.3)                                   
 gtable         0.3.0   2019-03-25 [2] CRAN (R 4.0.3)                                   
 haven          2.3.1   2020-06-01 [2] CRAN (R 4.0.3)                                   
 hms            1.0.0   2021-01-13 [2] CRAN (R 4.0.3)                                   
 httr           1.4.2   2020-07-20 [2] CRAN (R 4.0.3)                                   
 IRanges        2.24.1  2020-12-12 [2] Bioconductor                                     
 jsonlite       1.7.2   2020-12-09 [2] CRAN (R 4.0.3)                                   
 lifecycle      1.0.0   2021-02-15 [2] CRAN (R 4.0.4)                                   
 lubridate      1.7.9.2 2020-11-13 [2] CRAN (R 4.0.3)                                   
 magrittr       2.0.1   2020-11-17 [2] CRAN (R 4.0.3)                                   
 MASS           7.3-53  2020-09-09 [3] CRAN (R 4.0.4)                                   
 memoise        2.0.0   2021-01-26 [2] CRAN (R 4.0.3)                                   
 modelr         0.1.8   2020-05-19 [2] CRAN (R 4.0.3)                                   
 MsCoreUtils    1.3.3   2021-04-19 [1] Github (RforMassSpectrometry/MsCoreUtils@c490c07)
 munsell        0.5.0   2018-06-12 [2] CRAN (R 4.0.3)                                   
 pillar         1.4.7   2020-11-20 [2] CRAN (R 4.0.3)                                   
 pkgbuild       1.2.0   2020-12-15 [2] CRAN (R 4.0.3)                                   
 pkgconfig      2.0.3   2019-09-22 [2] CRAN (R 4.0.3)                                   
 pkgload        1.1.0   2020-05-29 [2] CRAN (R 4.0.3)                                   
 prettyunits    1.1.1   2020-01-24 [2] CRAN (R 4.0.3)                                   
 processx       3.4.5   2020-11-30 [2] CRAN (R 4.0.3)                                   
 ProtGenerics * 1.23.9  2021-04-19 [1] Github (lgatto/ProtGenerics@ff388f8)             
 ps             1.5.0   2020-12-05 [2] CRAN (R 4.0.3)                                   
 purrr        * 0.3.4   2020-04-17 [2] CRAN (R 4.0.3)                                   
 R6             2.5.0   2020-10-28 [2] CRAN (R 4.0.3)                                   
 Rcpp           1.0.6   2021-01-15 [2] CRAN (R 4.0.3)                                   
 readr        * 1.4.0   2020-10-05 [1] CRAN (R 4.0.3)                                   
 readxl         1.3.1   2019-03-13 [2] CRAN (R 4.0.3)                                   
 remotes        2.2.0   2020-07-21 [2] CRAN (R 4.0.3)                                   
 reprex         1.0.0   2021-01-27 [2] CRAN (R 4.0.3)                                   
 rlang          0.4.10  2020-12-30 [2] CRAN (R 4.0.3)                                   
 rprojroot      2.0.2   2020-11-15 [2] CRAN (R 4.0.3)                                   
 rstudioapi     0.13    2020-11-12 [2] CRAN (R 4.0.3)                                   
 rvest          0.3.6   2020-07-25 [2] CRAN (R 4.0.3)                                   
 S4Vectors    * 0.28.1  2020-12-09 [2] Bioconductor                                     
 scales         1.1.1   2020-05-11 [2] CRAN (R 4.0.3)                                   
 sessioninfo    1.1.1   2018-11-05 [2] CRAN (R 4.0.3)                                   
 Spectra      * 1.1.19  2021-04-19 [1] Github (rformassspectrometry/Spectra@ac3742b)    
 stringi        1.5.3   2020-09-09 [2] CRAN (R 4.0.3)                                   
 stringr      * 1.4.0   2019-02-10 [2] CRAN (R 4.0.3)                                   
 testthat       3.0.2   2021-02-14 [2] CRAN (R 4.0.4)                                   
 tibble       * 3.0.6   2021-01-29 [2] CRAN (R 4.0.3)                                   
 tidyr        * 1.1.2   2020-08-27 [2] CRAN (R 4.0.3)                                   
 tidyselect     1.1.0   2020-05-11 [2] CRAN (R 4.0.3)                                   
 tidyverse    * 1.3.0   2019-11-21 [2] CRAN (R 4.0.3)                                   
 usethis      * 2.0.1   2021-02-10 [2] CRAN (R 4.0.3)                                   
 vctrs          0.3.6   2020-12-17 [2] CRAN (R 4.0.3)                                   
 withr          2.4.1   2021-01-26 [2] CRAN (R 4.0.3)                                   
 xml2           1.3.2   2020-04-23 [2] CRAN (R 4.0.3)                                   

[1] C:/Archives_Libraries/R-4.0.4
[2] C:/Daten/msyshome/R/win-library/4.0
[3] C:/Program Files/R/R-4.0.4/library
```

**Comment by jorainer** [comment]
Thanks for reporting! I'll have a look at it.

**Comment by jorainer** [comment]
Fixed in version 1.1.20

**Comment by meowcat** [comment]
Can confirm.
